From 130482df6a6968b30b47be44cbdca34d6b39b598 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 9 Oct 2019 23:50:11 +0200
Subject: [PATCH 1/1] update for pacman 5.2 API change

With the release of pacman 5.2 we have an API change with upstream commit c0e9be79:

  commit c0e9be7973be6c81b22fde91516fb8991e7bb07b
  Author: Allan McRae <allan@archlinux.org>
  Date:   Sat Mar 2 18:57:20 2019 +1000

    Remove support for deltas from libalpm
---
 backends/alpm/pk-alpm-config.c      | 29 -----------------------------
 backends/alpm/pk-alpm-error.c       |  4 ----
 backends/alpm/pk-alpm-transaction.c | 19 -------------------
 configure.ac                        |  2 +-
 4 files changed, 1 insertion(+), 53 deletions(-)

diff --git a/backends/alpm/pk-alpm-config.c b/backends/alpm/pk-alpm-config.c
index 65843d0bd..c34224f0d 100644
--- a/backends/alpm/pk-alpm-config.c
+++ b/backends/alpm/pk-alpm-config.c
@@ -40,7 +40,6 @@ typedef struct
 {
 	 gboolean		 checkspace, color, ilovecandy, totaldl,
 				 usesyslog, verbosepkglists;
-	 gdouble		 deltaratio;
 
 	 gchar			*arch, *cleanmethod, *dbpath, *gpgdir, *logfile,
 				*root, *xfercmd;
@@ -65,7 +64,6 @@ pk_alpm_config_new (PkBackend *backend)
 {
 	PkAlpmConfig *config = g_new0 (PkAlpmConfig, 1);
 	config->backend = backend;
-	config->deltaratio = 0.0;
 
 	config->xrepo = g_regex_new ("\\$repo", 0, 0, NULL);
 	config->xarch = g_regex_new ("\\$arch", 0, 0, NULL);
@@ -147,14 +145,6 @@ pk_alpm_config_set_totaldl (PkAlpmConfig *config)
 	config->totaldl = TRUE;
 }
 
-static void
-pk_alpm_config_set_usedelta (PkAlpmConfig *config)
-{
-	g_return_if_fail (config != NULL);
-
-	config->deltaratio = 0.7;
-}
-
 static void
 pk_alpm_config_set_usesyslog (PkAlpmConfig *config)
 {
@@ -183,7 +173,6 @@ static const PkAlpmConfigBoolean pk_alpm_config_boolean_options[] = {
 	{ "Color", pk_alpm_config_set_color },
 	{ "ILoveCandy", pk_alpm_config_set_ilovecandy },
 	{ "TotalDownload", pk_alpm_config_set_totaldl },
-	{ "UseDelta", pk_alpm_config_set_usedelta },
 	{ "UseSyslog", pk_alpm_config_set_usesyslog },
 	{ "VerbosePkgLists", pk_alpm_config_set_verbosepkglists },
 	{ NULL, NULL }
@@ -292,22 +281,6 @@ pk_alpm_config_set_root (PkAlpmConfig *config, const gchar *path)
 	config->root = g_strdup (path);
 }
 
-static void
-pk_alpm_config_set_deltaratio (PkAlpmConfig *config, const gchar *number)
-{
-	gdouble ratio;
-	gchar *endptr;
-
-	g_return_if_fail (config != NULL);
-	g_return_if_fail (number != NULL);
-
-	ratio = g_ascii_strtod (number, &endptr);
-	/* this ignores invalid values whereas pacman reports an error */
-	if (*endptr == '\0' && 0.0 <= ratio && ratio <= 2.0) {
-		config->deltaratio = ratio;
-	}
-}
-
 static void
 pk_alpm_config_set_xfercmd (PkAlpmConfig *config, const gchar *command)
 {
@@ -333,7 +306,6 @@ static const PkAlpmConfigString pk_alpm_config_string_options[] = {
 	{ "GPGDir", pk_alpm_config_set_gpgdir },
 	{ "LogFile", pk_alpm_config_set_logfile },
 	{ "RootDir", pk_alpm_config_set_root },
-	{ "UseDelta", pk_alpm_config_set_deltaratio },
 	{ "XferCommand", pk_alpm_config_set_xfercmd },
 	{ NULL, NULL }
 };
@@ -947,7 +919,6 @@ pk_alpm_config_configure_alpm (PkBackend *backend, PkAlpmConfig *config, GError
 	alpm_option_set_checkspace (handle, config->checkspace);
 	alpm_option_set_usesyslog (handle, config->usesyslog);
 	alpm_option_set_arch (handle, config->arch);
-	alpm_option_set_deltaratio (handle, config->deltaratio);
 
 	/* backend takes ownership */
 	g_free (xfercmd);
diff --git a/backends/alpm/pk-alpm-error.c b/backends/alpm/pk-alpm-error.c
index 3547a48e9..31f1fab17 100644
--- a/backends/alpm/pk-alpm-error.c
+++ b/backends/alpm/pk-alpm-error.c
@@ -111,7 +111,6 @@ pk_alpm_error_emit (PkBackendJob *job, GError *error)
 	case ALPM_ERR_PKG_INVALID:
 	case ALPM_ERR_PKG_OPEN:
 	case ALPM_ERR_PKG_INVALID_NAME:
-	case ALPM_ERR_DLT_INVALID:
 		code = PK_ERROR_ENUM_INVALID_PACKAGE_FILE;
 		break;
 	case ALPM_ERR_PKG_INVALID_CHECKSUM:
@@ -126,9 +125,6 @@ pk_alpm_error_emit (PkBackendJob *job, GError *error)
 	case ALPM_ERR_SIG_MISSING:
 		code = PK_ERROR_ENUM_MISSING_GPG_SIGNATURE;
 		break;
-	case ALPM_ERR_DLT_PATCHFAILED:
-		code = PK_ERROR_ENUM_PACKAGE_FAILED_TO_BUILD;
-		break;
 	case ALPM_ERR_UNSATISFIED_DEPS:
 		code = PK_ERROR_ENUM_DEP_RESOLUTION_FAILED;
 		break;
diff --git a/backends/alpm/pk-alpm-transaction.c b/backends/alpm/pk-alpm-transaction.c
index e8e513229..c7e96dfee 100644
--- a/backends/alpm/pk-alpm-transaction.c
+++ b/backends/alpm/pk-alpm-transaction.c
@@ -62,15 +62,6 @@ pk_alpm_pkg_has_basename (PkBackend *backend, alpm_pkg_t *pkg, const gchar *base
 	if (g_strcmp0 (alpm_pkg_get_filename (pkg), basename) == 0)
 		return TRUE;
 
-	if (alpm_option_get_deltaratio (priv->alpm) == 0.0)
-		return FALSE;
-
-	for (i = alpm_pkg_get_deltas (pkg); i != NULL; i = i->next) {
-		alpm_delta_t *delta = (alpm_delta_t *) i->data;
-		if (g_strcmp0 (delta->delta, basename) == 0)
-			return TRUE;
-	}
-
 	return FALSE;
 }
 
@@ -647,7 +638,6 @@ pk_alpm_transaction_event_cb (alpm_event_t *event)
 	case ALPM_EVENT_RESOLVEDEPS_START:
 		pk_alpm_transaction_dep_resolve (job);
 		break;
-	case ALPM_EVENT_DELTA_INTEGRITY_START:
 	case ALPM_EVENT_DISKSPACE_START:
 	case ALPM_EVENT_FILECONFLICTS_START:
 	case ALPM_EVENT_INTERCONFLICTS_START:
@@ -700,10 +690,6 @@ pk_alpm_transaction_event_cb (alpm_event_t *event)
 	case ALPM_EVENT_LOAD_START:
 		pk_alpm_transaction_setup (job);
 		break;
-	case ALPM_EVENT_DELTA_PATCHES_START:
-	case ALPM_EVENT_DELTA_PATCH_START:
-		pk_alpm_transaction_repackaging (job);
-		break;
 	case ALPM_EVENT_SCRIPTLET_INFO:
 		pk_alpm_transaction_output (((alpm_event_scriptlet_info_t *) event)->line);
 		break;
@@ -726,10 +712,6 @@ pk_alpm_transaction_event_cb (alpm_event_t *event)
 		break;
 	case ALPM_EVENT_CHECKDEPS_DONE:
 	case ALPM_EVENT_DATABASE_MISSING:
-	case ALPM_EVENT_DELTA_INTEGRITY_DONE:
-	case ALPM_EVENT_DELTA_PATCH_DONE:
-	case ALPM_EVENT_DELTA_PATCHES_DONE:
-	case ALPM_EVENT_DELTA_PATCH_FAILED:
 	case ALPM_EVENT_DISKSPACE_DONE:
 	case ALPM_EVENT_FILECONFLICTS_DONE:
 	case ALPM_EVENT_HOOK_DONE:
@@ -1072,7 +1054,6 @@ pk_alpm_transaction_commit (PkBackendJob *job, GError **error)
 		alpm_list_free (data);
 		break;
 	case ALPM_ERR_PKG_INVALID:
-	case ALPM_ERR_DLT_INVALID:
 		prefix = pk_alpm_string_build_list (data);
 		alpm_list_free (data);
 		break;
diff --git a/configure.ac b/configure.ac
index 6de343b6a..9da8aea6f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -498,7 +498,7 @@ if test x$enable_aptcc = xyes; then
 fi
 
 if test x$enable_alpm = xyes; then
-	PKG_CHECK_MODULES(ALPM, libalpm >= 10.0.0)
+	PKG_CHECK_MODULES(ALPM, libalpm >= 12.0.0)
 fi
 
 if test x$enable_poldek = xyes; then
